test/common/common.go

```diff
@@ -251,35 +255,53 @@ func VerifyRegistryCache(parentCtx context.Context, log logr.Logger, shootClient
 	ctx, cancel = context.WithTimeout(parentCtx, 2*time.Minute)
 	defer cancel()
 
+	upstream, path, tag := splitImage(nginxImage)
 	selector := labels.SelectorFromSet(labels.Set(map[string]string{"upstream-host": strings.Replace(upstream, ":", "-", 1)}))
-	EventuallyWithOffset(1, ctx, func() (err error) {
-		reader, err := framework.PodExecByLabel(ctx, selector, "registry-cache", "cat /var/lib/registry/scheduler-state.json", metav1.NamespaceSystem, shootClient)
+	EventuallyWithOffset(1, ctx, func() error {
+		registryPod, err := framework.GetFirstRunningPodWithLabels(ctx, selector, metav1.NamespaceSystem, shootClient)
 		if err != nil {
-			return fmt.Errorf("failed to cat registry's scheduler-state.json file: %+w", err)
+			return fmt.Errorf("failed to get the registry pod: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to get a running registry Pod: %w", err)
```

As the `GetFirstRunningPodWithLabels` func does not only get the first running Pod but it also checks that the Pod is ready.